core/tsnclient/client.go

```diff
@@ -33,6 +33,7 @@ func NewClient(ctx context.Context, provider string, options ...Option) (*Client
 		return nil, err
 	}
 	c.kwilClient = kwilClient
+	c.Signer = kwilClient.Signer
```

Comment:
Nice catch, but I also see that the bug is that no error is being returned after the validation if it exists.

Could you also add this change to be complete?
```
	if err != nil {
		return nil, err
	}
```